### DISCUSSION TITLE
DataTable - Set current page programmatically

### DISCUSSION THREAD

**Discussion by mstimvol** [primefaces/community#1530] [PrimeVue] [answered]
I've a Dropdown and a DataTable in my template, e.g.:

```
Dropdown(v-model="typeId" inputId="type" filter :options="types" optionValue="id" optionLabel="name" :loading="loading.types" :disabled="loading.types" placeholder="Select plugin type you want to install")
DataTable(v-if="plugins.length" :loading="loading.plugins" v-model:selection="selectedPlugins" :value="plugins" paginator dataKey="id" :rows="5" :rowsPerPageOptions="[5, 10, 20, 50]" tableStyle="min-width: 50rem" @page="fnLoadPlugins" :totalRecords="totalPlugins" lazy)
```

The idea is, when I select an entry in the Dropdown, I load new data for the DataTable. Everything works fine, but when I'm on page 8 for example in DataTable and I change the selection in the Dropdown, it loads the new data. But in this case, there're only 2 pages available. In this case my DataTable becomes empty because page 8 is still open, although there're only 2 pages for the updated . So what I'm looking for is some kind of `setPage` for the DataTable, but I don't know how to do this. Any ideas?

I was also looking around but unfortunately found no solution.

**Comment by tugcekucukoglu** [comment, marked as answer]
You could try to use `first` property of DataTable.
